src/manage_sql_agent_groups.c

```diff
@@ -436,9 +436,23 @@ delete_agent_group (const char *agent_group_uuid, int ultimate)
            " SELECT %llu, agent_id FROM agent_group_agents WHERE group_id = %llu;",
            trash_id, agent_group);
 
+    /* Update the location of the agent_group in any trashcan tasks. */
```

Comment:
I think the moving the agent group to the trash should fail if the agent group is in use by a task that is not hidden/in the trash.